packages/jellyfish-api-core/__tests__/category/loan/updateVault.test.ts

```diff
@@ -0,0 +1,301 @@
+    // Update to different scheme and get liquidated
+    await testing.rpc.loan.updateVault(vaultId, {
+      ownerAddress: await testing.generateAddress(),
+      loanSchemeId: 'scheme'
+    })
+    await testing.generate(1)
+
+    {
+      const data = await testing.rpc.loan.getVault(vaultId)
+      expect(data.isUnderLiquidation).toStrictEqual(true)
+    }
+  })
+
+  it('should updateVault with loanSchemeId only', async () => {
+    await testing.rpc.loan.updateVault(createVaultId, {
+      loanSchemeId: 'scheme'
+    })
+    await testing.generate(1)
+
+    const data = await testing.rpc.loan.getVault(createVaultId)
+    expect(data.loanSchemeId).toStrictEqual('scheme')
+  })
+
+  it('should updateVault with owner address only', async () => {
+    const ownerAddress = await testing.generateAddress()
+    await testing.rpc.loan.updateVault(createVaultId, {
+      ownerAddress
+    })
+    await testing.generate(1)
+
+    const data = await testing.rpc.loan.getVault(createVaultId)
+    expect(data.ownerAddress).toStrictEqual(ownerAddress)
+  })
+
+  it('should not updateVault if vaultId is invalid', async () => {
+    const promise = testing.rpc.loan.updateVault('b25ab8093b0fcb712b9acecdb6ec21ae9cb862a14766a9fb6523efb1d8d05d60', {
+      ownerAddress: await testing.generateAddress(),
+      loanSchemeId: 'scheme'
+    })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Vault <b25ab8093b0fcb712b9acecdb6ec21ae9cb862a14766a9fb6523efb1d8d05d60> does not found\', code: -5, method: updatevault')
+  })
+
+  it('should not updateVault if the length of vaultId is not 64', async () => {
+    const promise = testing.rpc.loan.updateVault('x', {
+      ownerAddress: await testing.generateAddress(),
+      loanSchemeId: 'scheme'
+    })
+    await expect(promise).rejects.toThrow('RpcApiError: \'vaultId must be of length 64 (not 1, for \'x\')\', code: -8, method: updatevault')
+  })
+
+  it('should not updateVault if ownerAddress is invalid', async () => {
+    const promise = testing.rpc.loan.updateVault(createVaultId, {
+      ownerAddress: '1234'
+    })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Error: Invalid owner address\', code: -5, method: updatevault')
+  })
+
+  it('should not updateVault if loanSchemeId is invalid', async () => {
+    const promise = testing.rpc.loan.updateVault(createVaultId, {
+      loanSchemeId: '1234'
+    })
+
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test UpdateVaultTx execution failed:\nCannot find existing loan scheme with id 1234\', code: -32600, method: updatevault')
+  })
+
+  it('should not updateVault if owner address and loanSchemeId are not set', async () => {
+    const promise = testing.rpc.loan.updateVault(createVaultId, {})
+
+    await expect(promise).rejects.toThrow('RpcApiError: \'At least ownerAddress OR loanSchemeId must be set\', code: -8, method: updatevault')
+  })
+
+  // There is a bug in the c++ side
+  // it('should updateVault with utxos', async () => {
```

Comment:
I think for now, we are unable to updateVault with utxos. 
I believe this is a bug in the C++ side. Could reviewers please double check this for me?